examples/monitoring/cluster/main.cpp

```diff
@@ -14,77 +14,72 @@
  * limitations under the License.
  */
 #include <hazelcast/client/HazelcastClient.h>
-#include <hazelcast/client/InitialMembershipListener.h>
+#include <hazelcast/client/MembershipListener.h>
 #include <hazelcast/client/InitialMembershipEvent.h>
 #include <hazelcast/client/MembershipEvent.h>
 
-class MyInitialMemberListener : public hazelcast::client::InitialMembershipListener {
 
-public:
-    void init(hazelcast::client::InitialMembershipEvent event) override {
-        auto members = event.getMembers();
-        std::cout << "The following are the initial members in the cluster:" << std::endl;
-        for (const auto &member : members) {
-            std::cout << member.getAddress() << std::endl;
-        }
-    }
-
-    void memberAdded(const hazelcast::client::MembershipEvent &membershipEvent) override {
-        std::cout << "[MyInitialMemberListener::memberAdded] New member joined:" <<
-        membershipEvent.getMember().getAddress() << std::endl;
-    }
-
-    void memberRemoved(const hazelcast::client::MembershipEvent &membershipEvent) override {
-        std::cout << "[MyInitialMemberListener::memberRemoved] Member left:" <<
-        membershipEvent.getMember().getAddress() << std::endl;
-    }
-
-};
-
-class MyMemberListener : public hazelcast::client::MembershipListener {
-
-public:
-    void memberAdded(const hazelcast::client::MembershipEvent &membershipEvent) override {
-        std::cout << "[MyMemberListener::memberAdded] New member joined:" << membershipEvent.getMember().getAddress() <<
-        std::endl;
-    }
+MembershipListener makeMembershipListener() {
+    return MembershipListener()
+        .on_added([](const hazelcast::client::MembershipEvent &membershipEvent) {
+            std::cout << "[MyInitialMemberListener::memberAdded] New member joined:" <<
+            membershipEvent.getMember().getAddress() << std::endl;
+        })
+        .on_removed([](const hazelcast::client::MembershipEvent &membershipEvent) {
```

Comment:
should we better name these as `on_joined`, `on_left`? meaning that member joined to cluster, member left from cluster.